### PR TITLE
CGMES functional logs (first steps)

### PR DESCRIPTION
**Pull request by rcourtier**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**

Implements #3140

**What kind of change does this PR introduce?**

Feature

**What is the current behavior?**

There is some reporting done during CGMES import conversion but the logs are for the moment scattered and not hierarchically organized.

**What is the new behavior (if this is a feature change)?**

The purpose of that PR is to set a canvas of the main conversion steps in order to add the detailed functional logs.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [X] No

**Other information**:

### CODE REVIEW

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java [hunk 1]
```diff
@@ -141,130 +140,137 @@ public Network convert() {
...
+        convert(cgmes.energyConsumers(), CgmesNames.ENERGY_CONSUMER, context);
+        convert(cgmes.energySources(), CgmesNames.ENERGY_SOURCE, context);
+        convert(cgmes.equivalentInjections(), CgmesNames.EQUIVALENT_INJECTION, context);
+        convert(cgmes.externalNetworkInjections(), CgmesNames.EXTERNAL_NETWORK_INJECTION, context);
+        convert(cgmes.shuntCompensators(), CgmesNames.SHUNT_COMPENSATOR, context);
+        convert(cgmes.equivalentShunts(), CgmesNames.EQUIVALENT_SHUNT, context);
+        convert(cgmes.staticVarCompensators(), CgmesNames.STATIC_VAR_COMPENSATOR, context);
+        convert(cgmes.asynchronousMachines(), CgmesNames.ASYNCHRONOUS_MACHINE, context);
+        convert(cgmes.synchronousMachinesGenerators(), CgmesNames.SYNCHRONOUS_MACHINE, context);
+        convert(cgmes.synchronousMachinesCondensers(), "Condenser", context);
```
[the remark this answers is not in the document]
Comment (rcourtier) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java:199]: No, not like that. The second parameter is intended to be a CGMES class. The only CGMES class here is SynchronousMachine (CgmesNames.SYNCHRONOUS_MACHINE). The best solution would be to logwise merge the conversion of generators and condensers.

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java [hunk 2]
```diff
@@ -378,25 +384,40 @@ private static void addTieFlow(Context context, CgmesControlAreas cgmesControlAr
...
-                    logTitle = logTitle.replace("Conversion", "");
-                }
-                LOG.trace(element.tabulateLocals(logTitle));
+                LOG.trace(element.tabulateLocals(elementType));
             }
+            AbstractObjectConversion c = switch (elementType) {
+                case CgmesNames.SUBSTATION -> new SubstationConversion(element, context);
+                case CgmesNames.VOLTAGE_LEVEL -> new VoltageLevelConversion(element, context);
+                case CgmesNames.CONNECTIVITY_NODE -> new NodeConversion(CgmesNames.CONNECTIVITY_NODE, element, context);
+                case CgmesNames.TOPOLOGICAL_NODE -> new NodeConversion(CgmesNames.TOPOLOGICAL_NODE, element, context);
```
[the remark this answers is not in the document]
Comment (rcourtier) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java]: Done

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Context.java [hunk 3]
```diff
@@ -207,10 +208,22 @@ public PropertyBags phaseTapChangerTable(String tableId) {
 
     // Handling issues found during conversion
 
-    public ReportNode getReportNode() {
+    public ReportNode getRootReportNode() {
```
[the remark this answers is not in the document]
Comment (rcourtier) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Context.java]: Done

File: cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rcourtier) [reply cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Conversion.java]: Done

**Review by olperr1 — APPROVED** [review]

**Pull request merged**